### DISCUSSION TITLE
Can I intercept GraphQL API response?

### DISCUSSION THREAD

**Discussion by joelamltc** [directus/directus#16352] [Archive]
Is there a way to intercept GraphQL responses? I want to replace the error object with a standard message and log the error somewhere else. Thanks!

Refs: https://escape.tech/blog/graphql-error-handling-a-security-pov/

<img width="1297" alt="image" src="https://user-images.githubusercontent.com/13143636/200543915-29d93fe2-50ad-4c6a-acdb-9e61d5618e56.png">

**Comment by rijkvanzanten** [comment]
You can rely on hooks to intercept API traffic: https://docs.directus.io/extensions/hooks.html#custom-api-hooks

**Reply by joelamltc** [reply]
Yea, I have already went through it. However, I can't seem to find a way that can modify the GraphQL API response. Did I miss anything? It would be appreciated if you can give me some pointers. Thanks in advance!

**Reply by rijkvanzanten** [reply]
I've never tried adjusting the GQL response itself on the fly in a hook, so that'll require some experimentation 🙂

**Reply by joelamltc** [reply]
Yea, understood. The consideration is the error message from the GraphQL is too verbose which raises security concerns. We would like to hide it, log it in the server log only and not showing it to the client side.
